Source/Phone/Call/CallClient.swift

```diff
@@ -18,105 +18,134 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
+
 import Foundation
+import ObjectMapper
+
 
 // TODO: need to discuss if use term - "locus"
-class CallClient: CompletionHandlerType<CallInfo>{
+class CallClient {
+    
+    typealias ObjectHandler = (ServiceResponse<CallInfo>) -> Void
+    typealias ArrayHandler = (ServiceResponse<[CallInfo]>) -> Void
+    
+    private let authenticationStrategy: AuthenticationStrategy
+    private let deviceService: DeviceService
+    
+    init(authenticationStrategy: AuthenticationStrategy, deviceService: DeviceService) {
+        self.authenticationStrategy = authenticationStrategy
+        self.deviceService = deviceService
+    }
     
     private func requestBuilder() -> ServiceRequest.Builder {
-        return ServiceRequest.Builder().keyPath("locus")
+        return ServiceRequest.Builder(authenticationStrategy).keyPath("locus")
     }
     
-    func join(_ localInfo: RequestParameter, queue: DispatchQueue? = nil, completionHandler: @escaping ObjectHandler) {
+    private func body(deviceUrl: URL, json: [String:Any?] = [:]) -> RequestParameter {
+        var result = json
+        result["deviceUrl"] = deviceUrl.absoluteString
+        return RequestParameter(result)
+    }
+    
+    private func convertToJson(mediaInfo: MediaInfo) -> [String:Any?] {
+        let mediaInfoJSON = Mapper().toJSONString(mediaInfo, prettyPrint: true)!
+        return ["localMedias": [["type": "SDP", "localSdp": mediaInfoJSON]]]
+    }
+    
+    func createCall(toAddress: String, deviceUrl: URL, localMediaInfo: MediaInfo, queue: DispatchQueue? = nil, completionHandler: @escaping ObjectHandler) {
+        var json = convertToJson(mediaInfo: localMediaInfo)
+        json["invitee"] = ["address" : toAddress]
+
         let request = requestBuilder()
             .method(.post)
-            .baseUrl(DeviceService.sharedInstance.getServiceUrl("locus")!)
-            .body(localInfo)
+            .baseUrl(deviceService.device!.locusServiceUrl)
```

Comment:
Better than where it was before that! :D

This is one of those areas that will be much better served when there is proper management of registration state. CallClients could hopefully be injected with the registration data rather than querying for it. I feel like this is another area where we're 50 miles through a 100 mile journey.